src/index.js

```diff
@@ -1,44 +1,28 @@
 import {Bond, TimeBond, TransformBond as oo7TransformBond, ReactivePromise} from 'oo7';
 import BigNumber from 'bignumber.js';
-const Parity = require('@parity/parity.js');
 
 import { abiPolyfill, RegistryABI, RegistryExtras, GitHubHintABI, OperationsABI,
 	BadgeRegABI, TokenRegABI, BadgeABI, TokenABI } from './abis.js';
 
+import ParityApi from '@parity/api';
+
 // DEPRECATED. TODO: REMOVE
-export function setupBonds(_api = parity.api) {
+export function setupBonds(_api = ParityApi) {
 	return createBonds({ api: _api });
 }
 
-function injectedTransport() {
-	if (typeof(window) !== 'undefined' && typeof(window.parity) !== 'undefined' && window.parity.api.transport._url) {
-		return new Parity.Api.Transport.Http(
-			window.parity.api.transport._url[0] === '/'
-				? window.location.protocol + '//' + window.location.host + window.parity.api.transport._url
-			: window.parity.api.transport._url.contains('://')
-				? window.parity.api.transport._url
-				: window.location.href + window.parity.api.transport._url
-		);
-	}
-	return null;
-}
-
-function defaultTransport() {
-	var transport;
-
-	// Check to see if there's already a parity object injected in window.
-	transport = injectedTransport();
-
-	// Fallback to localhost:8545
-	if (!transport) {
-		transport = new Parity.Api.Transport.Http('http://localhost:8545');
-	}
-
-	return transport;
+function defaultProvider () {
+  	if(typeof window !== 'undefined' && window.ethereum){	
+  		return window.ethereum;
+  	}else if(typeof ParityApi !=='undefined'){
```

Comment:
Would keep the original top-level if as the else-if - that way it works both way around, i.e.

```
if (typeof window !== 'undefined' && window.ethereum) {
 ...
} else if (typeof(window) !== 'undefined' && typeof(window.parity) !== 'undefined' && window.parity.api.transport._url) {
}...
```

In addition typeof ParityApi will never be undefined, you include it.